signer/fulcio/github.go

```diff
@@ -0,0 +1,73 @@
+package fulcio
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/testifysec/go-witness/log"
+)
+
+const (
+	tokenAudience = "witness"
+	jwksURL       = "https://token.actions.githubusercontent.com/.well-known/jwks"
+)
+
+func fetchToken(tokenURL string, bearer string, audience string) (string, error) {
+	client := &http.Client{}
+
+	//add audient "&audience=witness" to the end of the tokenURL, parse it, and then add it to the query
+	u, err := url.Parse(tokenURL)
+	if err != nil {
+		return "", err
+	}
+
+	q := u.Query()
+	q.Add("audience", audience)
```

Comment:
if audience already exists in the tokenUrl passed to this function `q.Add` will add multiple audiences to the query string. is that desired? should we check if it exists and replace it if it does instead?